### PR TITLE
Have GUI output callback error on realization hover

### LINKED ISSUE #3578: Pass callback status as event and show in GUI
We now propagate a status message back from the callback: https://github.com/equinor/ert/blob/b2425049ae47c27a66ae24a75663a99b91db8dd4/res/job_queue/job_queue_node.py#L136 which is then logged and shown to the users through: https://github.com/equinor/ert/blob/65f1334060215f651408c18989613f3a010931da/ert_shared/models/base_run_model.py#L159 if all the simulations fail:

image

However if only a single realization fails, the user gets no feedback as to why. Would be good to propagate the message as an event instead and for example show the status information as hover info in the simulation tracker:

image

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #3578

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)
image

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/gui/model/node.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ class RealNodeData:
     real_status_color: Optional[QColor] = None
     current_memory_usage: Optional[int] = None
     max_memory_usage: Optional[int] = None
+    stderr: Optional[str] = None
```
Comment (berland) [inline src/ert/gui/model/node.py:83]: What happens if 10gigabytes is dumped to the stderr file?

Comment (jonathan-eq) [reply src/ert/gui/model/node.py:83]: :fire: Probably chaos...
I remember @xjules did a trick to only load a small part of the file if it is too large. I will see if I can find that again!

Comment (jonathan-eq) [reply src/ert/gui/model/node.py:83]: I set the max size to 1500 as this should only be seen as a preview of the error file.
image

**Comment by oyvindeide** [comment]
Think this looks like a good improvement! But dont think it solves the issue it is linked to? If there is an issue in the callback, that does not end up in the job error file, but is passed back as an event.

File: src/ert/gui/model/node.py (hunk 1 above)
Comment (berland) [reply src/ert/gui/model/node.py:83]: The lsf_driver has a function `read_stdout_and_stderr_files` which is related, but the driver might not be readily available to the gui.

**Comment by jonathan-eq** [comment]
> Think this looks like a good improvement! But dont think it solves the issue it is linked to? If there is an issue in the callback, that does not end up in the job error file, but is passed back as an event.

Oh, then I misunderstood the issue. Thanks for clarifying 👍

**Comment by xjules** [comment]
> > Think this looks like a good improvement! But dont think it solves the issue it is linked to? If there is an issue in the callback, that does not end up in the job error file, but is passed back as an event.
> 
> Oh, then I misunderstood the issue. Thanks for clarifying 👍

Yes, it should be https://github.com/equinor/ert/blob/6b7b21839ac89814da99e8e651b0271a3383e162/src/ert/scheduler/job.py#L248

**Comment by jonathan-eq** [comment]
image

File: src/ert/ensemble_evaluator/snapshot.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def __init__(self, snapshot: Optional["Snapshot"] = None) -> None:
         ] = defaultdict(dict)
         """A shallow dictionary of realization states. The key is a string with
         realization number, pointing to a dict with keys active (bool),
-        start_time (datetime), end_time (datetime) and status (str)."""
+        start_time (datetime), end_time (datetime), callback_status_message (str) and status (str)."""
```
Comment (xjules) [inline src/ert/ensemble_evaluator/snapshot.py:110]: I'd then emphasize the difference between these two statuses :)

File: src/ert/ensemble_evaluator/snapshot.py [hunk 3]
```diff
@@ -113,7 +113,9 @@ def __init__(self) -> None:
         ] = defaultdict(dict)
         """A shallow dictionary of realization states. The key is a string with
         realization number, pointing to a dict with keys active (bool),
-        start_time (datetime), end_time (datetime) and status (str)."""
+        start_time (datetime), end_time (datetime), callback_status_message (str) and status (str).
```
Comment (xjules) [inline src/ert/ensemble_evaluator/snapshot.py:116]: nice!

**Comment by xjules** [comment]
It looks very good! Just not sure if there should be a test for `callback_status_message`, maybe at least that the event contains `data["callback_status_message"]`?

**Review by xjules — APPROVED** [review]
Nice work @jonathan-eq! :rocket:

**Pull request merged**